### ISSUE TITLE
Parse messages indicating that a breakpoint has been enabled or disabled

### ISSUE THREAD

**Issue by cpitclaudel** [rocky/emacs-dbgr#102]
Although we do have support for displaying disabled breakpoints, we don't have support for parsing debugger messages signaling that a breakpoint has been disabled or enabled, do we?

If not, it would be a nice addition, I think :) But it's in parts of the codebase that I'm not too familiar with, so I'd welcome help with that.

**Comment by rocky** [comment]
Probably pretty easy to add for those debuggers that have enabling and disabling status messages. All of my debuggers do, but alas gdb doesn't and neither does pdb so I guess that extends to pdb variants.

First step is to add regular expression for parsing the enable and disable messages. For example:
https://github.com/rocky/emacs-dbgr/blob/master/realgud/debugger/trepan.pl/init.el#L127-L144

Then in _realgud-track-from-region_ you add this as a pattern to look for and take action on it. For example https://github.com/rocky/emacs-dbgr/blob/master/realgud/debugger/trepan.pl/init.el#L127-L144

**Comment by rocky** [comment]
commit e5a2331 in branch enable-disable does what I meant above. I'll need to include it in more debuggers, right now I tried in just bash. 

I was expecting a disabled breakpoint to be a hollow circle rather than a filled-in one. Is that correct or have I done something wrong?

**Comment by cpitclaudel** [comment]
This is really cool :) Let me look at the hollow circles thing.
Re multiple breakpoints: I wonder if it should be consistent with delete (should delete on a line with multiple breakpoints remove them all? Will we get confused if we get multiple messages from the debugger at once?). Alternatively, we could make sure that the "disable" command picks an enabled breakpoint (and conversely for the enable command), so that repeating "disable" disables all line breakpoints one by one.

**Comment by cpitclaudel** [comment]
Hmm, the breakpoint circle does turn hollow for me when I disable a breakpoint (I just tried in a fresh clone).  Are you not seeing this?

![screenshot from 2016-05-30 09 29 46](https://cloud.githubusercontent.com/assets/8181630/15650797/2f052356-2649-11e6-8a49-2a191f284095.png)

**Comment by rocky** [comment]
You are absolutely right. The gdb interface allows multiple breakpoints to be specified for enabling or disabling. And so that's what the POSIX shell (and all of my gdb-like debuggers) follow.

So I'll need to create a loop to look for all messages. 

There is no corresponding gdb command to disable or enable _all_ breakpoints in a line.

**Comment by cpitclaudel** [comment]
(bashdb is pretty cool, btw!)

**Comment by cpitclaudel** [comment]
> You are absolutely right. The gdb interface allows multiple breakpoints to be specified for enabling or disabling. And so that's what the POSIX shell (and all of my gdb-like debuggers) follow.

I see :) Though maybe a simpler solution is to disable just one breakpoint every time '-' is pressed (and enable just one when '+' is); we'd just have to be careful to pick a breakpoint that is in the right state. This could be simpler to implement, and it'd be fine interface-wise, I think.

**Comment by rocky** [comment]
With respect to the hollow arrows, what was going on is that I can't see it if the cursor also happens to be on that line. 
![bashdb-enabled](https://cloud.githubusercontent.com/assets/8851/15651081/3ad2db54-264b-11e6-8c75-019988ea20d1.png)
![bashdb-disabled](https://cloud.githubusercontent.com/assets/8851/15651083/3f8de80a-264b-11e6-8aa3-120f7d8ad478.png)

There is a hollow circle on line 4. Line 5 is for comparison.

**Comment by rocky** [comment]
I like disabling one breakpoint at a time with +/-. Of course the enable/disable commands have to still allow multiple enable/disable, so I'll need that loop. 

Yeah, basdhb is cool. But then so is the debugger for GNU make. And I like the the ability to deparse to find exact locations in the Perl and python debuggers. So it's all relative.

**Comment by rocky** [comment]
Hmm. Suppose the hollow circles had a red outline instead of grey? We need a style designer.

**Comment by cpitclaudel** [comment]
Yup, let's do red for all circles. I'll push that.

**Comment by cpitclaudel** [comment]
> Yeah, basdhb is cool. But then so is the debugger for GNU make. And I like the the ability to deparse to find exact locations in the Perl and python debuggers. So it's all relative. 

^^

**Comment by cpitclaudel** [comment]
Here's another possibility, which could make breakpoints more obvious:

![screenshot from 2016-05-30 10 24 35](https://cloud.githubusercontent.com/assets/8181630/15651936/baba6d14-2650-11e6-988a-af8afa94b10f.png)

This is using diff-removed-face, which users might customize, too.

**Comment by cpitclaudel** [comment]
The frame thing sounded good, but it adds two pixels to the line width; that isn't great. I'll look for something else.

**Comment by cpitclaudel** [comment]
Ok, alternative solution is to use rectangles by default in the fringe. I think I like this option:

![screenshot from 2016-05-30 10 45 36](https://cloud.githubusercontent.com/assets/8181630/15652372/c10cbbd8-2653-11e6-956f-f9e353da8cc9.png)

Opinions?

**Comment by rocky** [comment]
A design compromise. Fine with me!
